frontend/javascripts/oxalis/view/context_menu.tsx

```diff
@@ -467,132 +463,128 @@ function NodeContextMenuOptions({
-              onClick={() =>
-                activeNodeId != null
-                  ? extractShortestPathAsNewTree(clickedTree, activeNodeId, clickedNodeId)
-                  : null
+            : null,
+          {
+            key: "delete-edge",
+            disabled: !areNodesConnected,
+            onClick: () => (activeNodeId != null ? deleteEdge(activeNodeId, clickedNodeId) : null),
+            label: (
+              <>
+                Delete Edge to this Node{" "}
+                {useLegacyBindings ? shortcutBuilder(["Shift", "Ctrl", "leftMouse"]) : null}
+              </>
+            ),
+          },
+          {
+            key: "delete-node",
+            onClick: () => deleteNode(clickedNodeId, clickedTree.treeId),
+            label: (
+              <>
+                Delete this Node {activeNodeId === clickedNodeId ? shortcutBuilder(["Del"]) : null}
+              </>
+            ),
+          },
+          isBranchpoint
+            ? {
+                key: "branchpoint-node",
+                className: "node-context-menu-item",
+                onClick: () =>
+                  activeNodeId != null
+                    ? deleteBranchpointById(clickedNodeId, clickedTree.treeId)
+                    : null,
+                label: "Unmark as Branchpoint",
               }
-            >
-              Extract shortest Path to this Node
-            </Menu.Item>
-          )}
-        </>
-      ) : null}
-      {getMaybeMeshItems(
-        maybeClickedMeshId,
-        maybeMeshIntersectionPosition,
-        visibleSegmentationLayer,
-        datasetScale,
-        removeMesh,
-        hideMesh,
-        setPosition,
-        refreshMesh,
-      )}
-      {isTheSameNode ? null : (
-        <Menu.Item
-          key="measure-node-path-length"
-          disabled={activeNodeId == null || !areInSameTree || isTheSameNode}
-          onClick={() =>
+            : {
+                key: "branchpoint-node",
+                className: "node-context-menu-item",
+                onClick: () =>
+                  activeNodeId != null
+                    ? createBranchPoint(clickedNodeId, clickedTree.treeId)
+                    : null,
+                label: (
+                  <>
+                    Mark as Branchpoint{" "}
+                    {activeNodeId === clickedNodeId ? shortcutBuilder(["B"]) : null}
+                  </>
+                ),
+              },
+          isTheSameNode
+            ? null
+            : {
+                key: "extract-shortest-path",
+                disabled: activeNodeId == null || !areInSameTree || isTheSameNode,
+                onClick: () =>
+                  activeNodeId != null
+                    ? extractShortestPathAsNewTree(clickedTree, activeNodeId, clickedNodeId)
+                    : null,
+                label: "Extract shortest Path to this Node",
+              },
+        ]
+      : []),
+    ...(maybeMeshItems ? maybeMeshItems : []),
```

Comment:
would probably be nicer to have `getMaybeMeshItems` return an empty array instead of null so that the distinction here doesn't have to be made. then, it could also be renamed to not contain the "maybe" in the function and var name.